test/executor.test.js

```diff
@@ -906,6 +906,43 @@ describe('executor', function() {
     });
   });
 
+  describe.only('when booting with lazy connect', function() {
+    var DATASOURCE_THROWING_CONNECTION_ERROR = {
+      lazyConnector: {
+        connector: 'testLazyConnect',
+        name: 'lazyConnector',
+      },
+    };
+    var connectTriggered = true;
+
+    beforeEach(function() {
+      app.connector('testLazyConnect', {
+        initialize: function(dataSource, callback) {
+          if (dataSource.settings.lazyConnect) {
+            connectTriggered = false;
+          }
+        },
+      });
+    });
+
+    it('should trigger connect', function(done) {
+      boot.execute(app, someInstructions({
+        dataSources: DATASOURCE_THROWING_CONNECTION_ERROR,
+      }));
```

Comment:
To avoid repetition, please extract `someInstructions({ dataSources: DATASOURCE_THROWING_CONNECTION_ERROR }); ` into a shared constant. Then you can remove `DATASOURCE_THROWING_CONNECTION_ERROR` entirely (inline it in the new constant).